lib/jobs/pdu-node-relations.js

```diff
@@ -0,0 +1,191 @@
+            return self._searchObmServices(pduHost, node);
+        })
+        .then(function () {
+            self._done();
+        })
+        .catch(function (err) {
+            self._done(err);
+        });
+
+    };
+
+    PduRelationsJob.prototype._searchObmServices =
+        function _searchObmServices(pduHost, pduNode) {
+            var self = this;
+            return Promise.try(function () {
+                return waterline.obms.find();
+            })
+            .filter(function (obm) {
+                return (_validateIpAddressFormat(obm.config.host));
+            })
+            .map(function (result) {
+                /* Compare hosts
+                 * If PDu host and the pdu obm service host are equal
+                 * Then create the relation between node and discoverd pdu
+                */
+                var obmServiceHost = _adjustIpStrFormat(result.config.host);
+                if (pduHost === obmServiceHost) {
+                    //find the node
+                    return self.updatePoweredNodeRelations(result.node)
+                   .then(function () {
+                       return self.updatePduNodeRelations(pduNode, result.node);
+                   });
+                }
+            });
+        };
+
+    PduRelationsJob.prototype.updatePoweredNodeRelations =
+        function _updatePoweredNodeRelations(poweredNodeId) {
+            var self = this;
+            return waterline.nodes.findByIdentifier(poweredNodeId)
+            .then(function (node) {
+                if (!node) {
+                    throw new Error('Could not find node with identifier ' + node);
+                }
+                return node.relations;
+            })
+            .filter(function (entry) {
+                return (entry.targets.indexOf(self.nodeId) < 0);
+            })
+            .then(function (relations) {
+                relations.push({
+                    relationType: NODE_RELATION_TYPE,
+                    targets: [self.nodeId]
+                });
+                return waterline.nodes.updateByIdentifier(
+                    poweredNodeId,
+                    { relations: relations });
+            });
+        };
+
+    PduRelationsJob.prototype.updatePduNodeRelations =
+        function _updatePduNodeRelations(pduNode, poweredNodeId) {
+            Promise.try(function () {
+                return pduNode.relations;
+            })
+            .filter(function (entry) {
+                //return entries except the one with relation type = powers
+                return (entry.relationType !== PDU_RELATION_TYPE);
+            })
+            .then(function (pduRelations) {
+                //update PDU relation type powers
+                var targets = _getTargets(pduNode, PDU_RELATION_TYPE);
+                if (targets.indexOf(poweredNodeId) < 0) {
+                    targets.push(poweredNodeId);
+                }
+                pduRelations.push({
```

Comment:
If the pduRelations is empty, you can push a new `powers` relation record. But if the pudRelations is not empty, that means you don't need to create a new relation record, just update the existing relation record and insert the new nodeId to targets is enough.